spec/queries/comments/community_wellness_query_spec.rb

```diff
@@ -23,8 +23,45 @@
     create_comment_time_ago(user4.id, 239.days.ago, commentable: articles.sample)
   end
 
+  # This context spec will likely start to fail when we are ready to remove the
+  # rollout limitation from the query+logic. Read more about this here:
+  #
+  # - https://github.com/forem/forem/issues/17310#issuecomment-1118554640
+  # - https://github.com/forem/forem/blob/main/app/queries/comments/community_wellness_query.rb#L28
+  context "when the current date is less than (precedes) December 19, 2022" do
+    it "uses the date-limited query" do
+      expect(described_class.limit_query_rollout?).to be true
+    end
+
+    it "returns a max number of weeks that matches the diff from rollout date" do
+      user6 = create(:user)
+
+      # Add 2 comments per week (including week 0) for `user6`
+      (0..33).each do |i|
+        num_days_ago = (2 + (i * 7)).days.ago
+        create_comment_time_ago(user6.id, num_days_ago, commentable: articles.sample)
+        create_comment_time_ago(user6.id, num_days_ago, commentable: articles.sample)
+      end
+
+      # Fetch and de-serialize the results for `user6`
+      result = described_class.call
+      index6 = result.index { |hash| hash["user_id"] == user6.id }
+      weeks_ago_array = result[index6]["serialized_weeks_ago"].split(",")
+      comment_counts_array = result[index6]["serialized_comment_counts"].split(",")
+
+      # If `1.34` weeks have passed we should get no more than 2 weeks in the
+      # result arrays because of the `limit_release_date_sql_query`. That's what
+      # is being calculated here and stored in `expected_weeks`
+      expected_weeks = ((Time.current - Time.zone.parse("2022-05-01")) / 7.days).ceil
+      expect(weeks_ago_array.count).to eq(expected_weeks)
+      expect(comment_counts_array.count).to eq(expected_weeks)
+    end
+  end
```

Comment:
This spec will check, week by week, that the query is still using the SQL query that limits the number of weeks returned based on the diff with the rollout date. Both of these specs (from this same `context`) will likely start to fail when 33 weeks have passed (mid-December). However, as mentioned before I'm hoping to leave a reminder for the internal team to handle this in a quick and easy way.